### DISCUSSION TITLE
Background Image

### DISCUSSION THREAD

**Discussion by omarraja786** [QuestPDF/QuestPDF#175] [Q&A] [answered]
Hi there,

I am creating an invoice and I was wondering if it is possible to set the background as an image. I know it supports a hex colour string or Colors  but I'm not sure if there is a way to set a custom background. 

Or is it possible to have a work around i.e. having an image that stretches across the whole page and having all other content on top of it?

Thank you,
Omar

**Comment by MarcinZiabek** [comment, marked as answer]
Hello 😁 

You can use [the Layers element](https://www.questpdf.com/documentation/api-reference.html#layers) to put an image behind or in front of your content. If the background image should work as a watermark, you can also utilize proper slots in [the Page element](https://www.questpdf.com/documentation/api-reference.html#watermark-slots). Just make sure that your image uses proper aspect ratio and has enough space, so to avoid the InfiniteDocument exception

Please notice that both solutions accept any container and content, not only images. So technically, your background can consists of a table with text 🤣

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
